Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -3472,7 +3473,11 @@ else if (thisReq_ instanceof ServerRequestLogout) {
                 } catch (JSONException ex) {
                     ex.printStackTrace();
                 }
+                if (thisReq_ instanceof ServerRequestInitSession) {
+                    branchUniversalReferralInitWrapper_ = null;
```

Comment:
The CallbackWrapper is nullified only on receiving install/open response and only after calling  `onInitFinished`.  So once callback set it will get nullified only after calling `onInitFinished`